### PR TITLE
Fix `/<dojo>/modules` API endpoint

### LINKED ISSUE #527: `/<dojo>/modules` API endpoint doesn't work
Calling the endpoint results in Internal Server Error
For instance:
https://pwn.college/pwncollege_api/v1/dojo/fundamentals/modules

https://github.com/pwncollege/dojo/blob/59b31c64b010f3b4b3f1e70fefc77d832d1a8882/dojo_plugin/api/v1/dojo.py#L155-L166

### PR DESCRIPTION
**Pull request by fpotier**
Should close #527

### CODE REVIEW

File: dojo_plugin/models/__init__.py [hunk 1]
```diff
@@ -405,6 +405,14 @@ def resources(self, value):
             resource.resource_index = resource_index
         self._resources = value
 
+    @hybrid_method
+    def visible(self):
+        return len(self.visible_challenges()) > 0
+
+    @visible.expression
+    def visible(cls, when=None):
+        return len(cls.visible_challenges()) > 0
```
Comment (ConnorNelson) [inline dojo_plugin/models/__init__.py:414]: Are you sure that this is valid sqlalchemy? At a glance it doesn't look like it would be since it iterates.

Comment (fpotier) [reply dojo_plugin/models/__init__.py:414]: I'm not, tbh I'm not familiar with sqlachemy. I just copied the `visible()` method of `DojoChallenges`

**Comment by ConnorNelson** [comment]
Thanks for the testcase. I think this is the correct patch though. A module is still considered visible even if it doesn't have any visible challenges I believe (for example to see lecture materials, etc).

**Comment by fpotier** [comment]
That makes sense, I didn't think about this case. Thank you

**Comment by fpotier** [comment]
@ConnorNelson is there any case where a module should not/no longer be visible that would require another test case?

**Comment by ConnorNelson** [comment]
I don't think so. Maybe eventually we'll want to support something like that, but for now this looks great. Thanks for the PR!

**Pull request merged**